### PR TITLE
Clear button added

### LINKED ISSUE #437: Clear button in search bar
The search image in search bar is not clickable so it serves no purpose. It can be replaced by a clear button that could clear the search query easily.

Screenshot from 2019-03-15 13-23-49

### PR DESCRIPTION
**Pull request by richa208**
Fixes #437 

hot

@smit1678 @ramyaragupathy Kindly review this.

### CODE REVIEW

File: _sass/base.scss [hunk 1]
```diff
@@ -1721,15 +1721,17 @@ em {
   color: $blue-dark;
   box-shadow: none;
   flex: 1;
-  background-image: url(../images/icon-search.svg);
```
Comment (ramyaragupathy) [inline _sass/base.scss:1724]: Try retaining this block and play around a little with opacity or jquery attribute fetch

File: js/main.js [hunk 2]
```diff
@@ -53,6 +53,16 @@ $(".search-close").on("click", function() {
   $('body').removeClass('search-on');
 });
 
+$("#search-input").on("input", function() {
+  var content = $(this).val();
+  $('#search-input').css({'background-image' : '', 'background-repeat' : '', 'background-size':'', 'background-position': ''});  
+  if(content=='')
+  {
+    $('#search-input').css({'background-image' : 'url(../images/icon-search.svg)', 'background-repeat' : 'no-repeat', 'background-size':'18px 18 px', 'background-position': 'right 14px center'});
```
Comment (ramyaragupathy) [inline js/main.js]: Please avoid inline styles as it leads to code redundancy. Try playing around with opacity or jquery attribute fetch

File: js/main.js [hunk 3]
```diff
@@ -53,6 +53,16 @@ $(".search-close").on("click", function() {
   $('body').removeClass('search-on');
 });
 
+$("#search-input").on("input", function() {
+  var content = $(this).val();
+  $('#search-input').css({'background-image' : '', 'background-repeat' : '', 'background-size':'', 'background-position': ''});
```
Comment (ramyaragupathy) [inline js/main.js]: Please avoid inline styles as it leads to code redundancy. Try playing around with opacity or jquery attribute fetch

**Comment by richa208** [comment]
@ramyaragupathy Suggested changes have been done.
Kindly review it and suggest if it needs improvement.

**Comment by ramyaragupathy** [comment]
ok - this works well in Chrome, but the cancel button doesn't show up in Firefox

**Review by ramyaragupathy — CHANGES_REQUESTED** [review]
From the [documentation for webkit cancel button](https://developer.mozilla.org/en-US/docs/Web/CSS/::-webkit-search-cancel-button) - this is not a standard feature and hence will not work on every browser. There is no point digging into this further @richa208 in this direction. Suggest you think of other design solutions here @richa208

**Comment by richa208** [comment]
Yes @ramyaragupathy I also found out that it works on selected browsers only. 
Will update the PR with appropriate solution soon.

**Comment by richa208** [comment]
@ramyaragupathy I made some changes in the UI by adding a clear button as I was not able to attach a click event on the background image.

hot4

Kindly tell if it needs more improvement.

**Comment by ramyaragupathy** [comment]
Not completely convinced about having a button outside as I don't find it adding to the user experience. I suggest let's close it here @richa208 and may be you could post other improvised search design options in your ticket. Let's not rush for any quick fix as it not a pressing problem or a bug.

**Comment by richa208** [comment]
Sure @ramyaragupathy as you find it better for the website. 
I will post other design options in the ticket if they are appropriate.

**Comment by ramyaragupathy** [comment]
Thanks @richa208 - closing this here now

**Pull request closed without merging**